relay-event-normalization/src/normalize/span/tag_extraction.rs

```diff
@@ -1050,6 +1052,27 @@ pub fn extract_tags(
                 if let Some(release) = span.data.value().and_then(|data| data.release.as_str()) {
                     span_tags.release = release.to_owned().into();
                 }
+                // Standalone vital spans don't come from an event with geo data.
+                // Derive geo data from the client address inferred on the span.
+                if let Some(client_address) = span
+                    .data
+                    .value()
+                    .and_then(|data| data.client_address.value())
+                {
+                    if let Some(geoip_lookup) = geoip_lookup {
+                        if let Ok(Some(geo)) = geoip_lookup.lookup(client_address.as_str()) {
+                            if let Some(country_code) = geo.country_code.value() {
+                                span_tags.user_country_code = country_code.to_owned().into();
+                                if let Some(subregion) = Subregion::from_iso2(country_code.as_str())
+                                {
+                                    let numerical_subregion = subregion as u8;
+                                    span_tags.user_subregion =
+                                        numerical_subregion.to_string().into();
+                                }
+                            }
+                        }
+                    }
+                }
```

Comment:
With let chains being released in Rust 1.88.0, this could be written with way less nesting:

```suggestion
                if let Some(client_address) = span
                    .data
                    .value()
                    .and_then(|data| data.client_address.value())
                    && let Some(geoip_lookup) = geoip_lookup
                    && let Ok(Some(geo)) = geoip_lookup.lookup(client_address.as_str())
                    && let Some(country_code) = geo.country_code.value()
                {
                    span_tags.user_country_code = country_code.to_owned().into();
                    if let Some(subregion) = Subregion::from_iso2(country_code.as_str()) {
                        let numerical_subregion = subregion as u8;
                        span_tags.user_subregion = numerical_subregion.to_string().into();
                    }
                }
```

However, I don't know if we want to make Relay require the most recent version of the Rust compiler. @Dav1dde wdyt?

Reply:
@loewenheim We mention this in [README](https://github.com/getsentry/relay?tab=readme-ov-file#development) (see first sentence) that we require the latest stable version of Rust.

Reply:
Oh, perfect :)

Reply:
Thanks! Definitely an improvement!